components/canvas/canvas_data.rs

```diff
@@ -0,0 +1,969 @@
+                                   self.state.stroke_opts.mDashPattern);
+            self.drawtarget.stroke_line(rect.origin, rect.bottom_right(),
+                                        self.state.stroke_style.to_pattern_ref(),
+                                        &stroke_opts, &self.state.draw_options);
+        } else {
+            self.drawtarget.stroke_rect(rect, self.state.stroke_style.to_pattern_ref(),
+                                        &self.state.stroke_opts, &self.state.draw_options);
+        }
+    }
+
+    pub fn begin_path(&mut self) {
+        self.path_builder = self.drawtarget.create_path_builder()
+    }
+
+    pub fn close_path(&self) {
+        self.path_builder.close()
+    }
+
+    pub fn fill(&self) {
+        if is_zero_size_gradient(&self.state.fill_style) {
+            return; // Paint nothing if gradient size is zero.
+        }
+
+        self.drawtarget.fill(&self.path_builder.finish(),
+                             self.state.fill_style.to_pattern_ref(),
+                             &self.state.draw_options);
+    }
+
+    pub fn stroke(&self) {
+        if is_zero_size_gradient(&self.state.stroke_style) {
+            return; // Paint nothing if gradient size is zero.
+        }
+
+        self.drawtarget.stroke(&self.path_builder.finish(),
+                               self.state.stroke_style.to_pattern_ref(),
+                               &self.state.stroke_opts,
+                               &self.state.draw_options);
+    }
+
+    pub fn clip(&self) {
+        self.drawtarget.push_clip(&self.path_builder.finish());
+    }
+
+    pub fn is_point_in_path(&mut self, x: f64, y: f64,
+                        _fill_rule: FillRule, chan: IpcSender<bool>) {
+        let path = self.path_builder.finish();
+        let result = path.contains_point(x, y, &self.state.transform);
+        self.path_builder = path.copy_to_builder();
+        chan.send(result).unwrap();
+    }
+
+    pub fn move_to(&self, point: &Point2D<AzFloat>) {
+        self.path_builder.move_to(*point)
+    }
+
+    pub fn line_to(&self, point: &Point2D<AzFloat>) {
+        self.path_builder.line_to(*point)
+    }
+
+    pub fn rect(&self, rect: &Rect<f32>) {
+        self.path_builder.move_to(Point2D::new(rect.origin.x, rect.origin.y));
+        self.path_builder.line_to(Point2D::new(rect.origin.x + rect.size.width, rect.origin.y));
+        self.path_builder.line_to(Point2D::new(rect.origin.x + rect.size.width,
+                                               rect.origin.y + rect.size.height));
+        self.path_builder.line_to(Point2D::new(rect.origin.x, rect.origin.y + rect.size.height));
+        self.path_builder.close();
+    }
+
+    pub fn quadratic_curve_to(&self,
+                          cp: &Point2D<AzFloat>,
+                          endpoint: &Point2D<AzFloat>) {
+        self.path_builder.quadratic_curve_to(cp, endpoint)
+    }
+
+    pub fn bezier_curve_to(&self,
+                       cp1: &Point2D<AzFloat>,
```

Comment:
And in a bunch of these, please format them using the common style:

```rust
pub fn bezier_curve_to(
    &self,
    cp1: &Point2D<AzFloat>,
    // ...
) {
```

Reply:
Thanks, I wasn't quite sure what the common style was since it was kinda inconsistent. I'm more than happy to move all this over to that, though.